packages/codemod/src/utils/prop-utils.ts

```diff
@@ -0,0 +1,19 @@
+import { ASTPath, JSXOpeningElement } from "jscodeshift";
+
+export function updatePropName(
+  elementPath: ASTPath<JSXOpeningElement>,
+  propsNamesMappingOldToNew: Record<string, string>
+): void {
+  const attributes = elementPath.node?.attributes;
+  if (!attributes) return;
+
+  attributes.forEach(attr => {
+    if (
+      attr.type === "JSXAttribute" &&
+      attr.name.type === "JSXIdentifier" &&
+      propsNamesMappingOldToNew[attr.name.name]
+    ) {
+      attr.name.name = propsNamesMappingOldToNew[attr.name.name];
```

Comment:
What will happen if for some reason the component already using the new prop in addition to old one? for example:
< AttentionBox componentClassName="old-class" className="other classes" />

Reply:
Not sure how this code will resolve the use case I wrote..

Reply:
you asked what will happen, not how to resolve 😅 
I don't think we should do anything in this case
TS would yell at the consumer that `componentClassName`, and if the consumer does not have TS configured it is its responsibility to check the docs when migrating 🤷🏼‍♂️

If we decide to remove the `componentClassName` no matter what is its content in case `className` also exists, then the user would most probably get hurt (unexpected and unaware changes)

Reply:
sorry, fixed the snippet, was probably confusing

see the fixed one

it is also already committed so you can just go over the commit

Reply:
![image](https://github.com/mondaycom/vibe/assets/30905362/135f392a-3d7b-456d-9b5f-206398242d75)

```typescript
elements.forEach(path => {
    const hasClassName = isPropExists(j, path, "className");
    const hasComponentClassName = isPropExists(j, path, "componentClassName");
    if (hasClassName && hasComponentClassName) {
      logPropMigrationWarning(filePath, componentName, "componentClassName", "className");
    } else {
      updatePropName(path, { componentClassName: "className" });
    }
  });
```